src/plugins/github/fetchReposForOrg.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+
+import {type RepoId, makeRepoId} from "../../core/repoId";
+import dedent from "../../util/dedent";
+
+const GITHUB_GRAPHQL_SERVER = "https://api.github.com/graphql";
+async function ghquery(query, token): any {
+  const body = JSON.stringify({query});
+  const fetchOptions = {
+    method: "POST",
+    body,
+    headers: {
+      Authorization: `bearer ${token}`,
+    },
+  };
+  const response = await fetch(GITHUB_GRAPHQL_SERVER, fetchOptions);
+  const json = await response.json();
+  if (json.errors) {
+    throw json.errors;
+  }
+  return json;
+}
+
+export async function fetchReposForOrg(
+  org: string,
+  token: string
+): Promise<$ReadOnlyArray<RepoId>> {
+  const query = dedent`\
+  {
+    search(query: "org:${org}", type: REPOSITORY, first: 50) {
+      nodes {
+        ... on Repository {
+          name
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+  `;
```

Comment:
Please prefer the structured API over manual string interpolation:

```js
  const b = Queries.build;
  const payload = {
    body: [
      b.query(
        "PerformSearch",
        [b.param("searchQuery", "String!")],
        [
          b.field(
            "search",
            {
              query: b.variable("searchQuery"),
              type: b.literal("REPOSITORY"),
              first: b.literal("30"),
            },
            [
              b.field("nodes", {}, [
                b.inlineFragment("Repository", [b.field("name")]),
              ]),
              b.field("pageInfo", {}, [
                b.field("endCursor"),
                b.field("hasNextPage"),
              ]),
            ]
          ),
        ]
      ),
    ],
    variables: {searchQuery: `org:${org}`},
  };
  const result = await postQuery(payload);
```

This emits

```graphql
query PerformSearch($searchQuery: String!) {
  search(query: $searchQuery type: "REPOSITORY" first: "30") {
    nodes {
      ... on Repository {
        name
      }
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}
```

just like your version, but it is guaranteed at compile time that the
query is structurally correct (e.g., you have the correct number of
closing braces, everything is quoted properly, GraphQL syntax is
correct). Also, this sends fewer bytes over the wire because the
`inlineLayout` version is

```
query PerformSearch($searchQuery: String!) { search(query: $searchQuery type: "REPOSITORY" first: "30") { nodes { ... on Repository { name } } pageInfo { endCursor hasNextPage } } }
```

which is only 77% of the size of the original.

Reply:
Unfortunately, what your version emits is _not_ the same as what my version emits. In particular, my version emits:

```

Reply:
Update: I've realized this is covered by the ENUM use case.

Reply:
Er, yes, sorry, that should have been `b.enumLiteral`. My bad.